### PR TITLE
Adds a `get_production_properties()` method to the Python interface to Opm::Schedule

### PR DESCRIPTION
**Pull request by hakonhagland**
See use case downstream: https://github.com/OPM/opm-simulators/pull/3910
Adds a method `get_production_properties()` to the Python interface to `Opm::Schedule`. This method can be used to get information about `Well::WellProductionProperties` for a given report step.

### CODE REVIEW

**Comment by hakonhagland** [comment]
jenkins build this please

File: python/cxx/schedule.cpp [hunk 1]
```diff
@@ -56,6 +58,57 @@ namespace {
...
+            }
+            {
+                auto rate = prod_prop.THPTarget.get<double>();
+                props.insert(std::make_pair("thp_target", rate));
+            }
+            {
+                auto rate = prod_prop.ALQValue.get<double>();
+                props.insert(std::make_pair("alq_value", rate));
+            }
+            return props;
```
Comment (bska) [inline python/cxx/schedule.cpp:105]: Any reason not to write this as
```C++
return {
        { "oil_rate", prod_prop.OilRate.get<double>() },
        { "gas_rate", prod_prop.GasRate.get<double>() },
        // ...
        { "alq_value", prod_prop.ALQValue.get<double>() },
    };
```

Comment (hakonhagland) [reply python/cxx/schedule.cpp]: @bska Thanks that looks much nicer yes! I have updated the PR.

**Comment by hakonhagland** [comment]
jenkins build this please

File: python/cxx/schedule.cpp [hunk 2]
```diff
@@ -68,46 +68,21 @@ namespace {
...
-            return props;
+            return {
+                { "oil_rate", prod_prop.OilRate.get<double>() },
+                { "gas_rate", prod_prop.GasRate.get<double>() },
+                { "water_rate", prod_prop.WaterRate.get<double>() },
+                { "liquid_rate", prod_prop.LiquidRate.get<double>() },
+                { "resv_rate", prod_prop.ResVRate.get<double>() },
+                { "bhp_target", prod_prop.BHPTarget.get<double>() },
+                { "thp_target", prod_prop.THPTarget.get<double>() },
+                { "alq_value", prod_prop.ALQValue.get<double>() }
```
Comment (bska) [inline python/cxx/schedule.cpp:80]: Thanks for the update.  For what it's worth, you're allowed&mdash;some might even say encouraged&mdash;to add a trailing comma to the last element of the initialiser list here.  Having the final comma makes the list more uniform and slightly easier to update if you need to add more items or remove some existing item.

Comment (hakonhagland) [reply python/cxx/schedule.cpp]: Thanks for the information. Good to know, I have updated the PR again.

**Review by bska — APPROVED** [review]
Thanks for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**